src/hotspot/os/posix/signals_posix.cpp

```diff
@@ -118,6 +118,7 @@ class SavedSignalHandlers {
     if (check_signal_number(sig)) {
       assert(_sa[sig] == NULL, "Overwriting signal handler?");
       _sa[sig] = NEW_C_HEAP_OBJ(struct sigaction, mtInternal);
+      *_sa[sig] = *act;
```

Comment:
Sorry I missed the fact we weren't actually copying the incoming sigaction struct!

Reply:
Yes me too, and I'm annoyed by the fact that no tier1 test did catch that. The full jtreg suite would have catched it but it did not run yet since the last commit.